service/history/ndc/workflow_resetter.go

```diff
@@ -137,6 +138,10 @@ func (r *workflowResetterImpl) ResetWorkflow(
 	var reapplyEventsFn workflowResetReapplyEventsFn
 	currentMutableState := currentWorkflow.GetMutableState()
 	if currentMutableState.IsWorkflowExecutionRunning() {
+		currentMutableState.GetExecutionInfo().WorkflowReset = true
+		if !currentMutableState.GetExecutionInfo().WorkflowReset {
```

Comment:
You just set this, why do you need this check?

Reply:
This is just a draft. Extra debugging logs. Will cleanup